nearcore/src/state_sync.rs

```diff
@@ -264,7 +314,74 @@ async fn state_sync_dump(
                         None
                     }
                     Ok((state_root, num_parts, sync_prev_prev_hash)) => {
-                        match get_missing_part_ids_for_epoch(
+                        // Upload header
+                        let header_in_external_storage =
+                            match is_state_sync_header_missing_for_epoch(
+                                shard_id,
+                                &chain_id,
+                                &epoch_id,
+                                epoch_height,
+                                &external,
+                            )
+                            .await
+                            {
+                                Err(err) => {
+                                    tracing::error!(target: "state_sync_dump", ?err, ?shard_id, "Failed to determine header presence in external storage.");
+                                    false
+                                }
+                                // Header is already stored
+                                Ok(false) => true,
```

Comment:
super nit: change `is_state_sync_header_missing_for_epoch` to `is_state_sync_header_stored_for_epoch` to make it `Ok(true) => true` which is more consistent